script.module.youtube.dl/lib/main.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+import sys
+from lib.yd_private_libs import util, servicecontrol, updater
+import xbmc
+import xbmcgui
+
+
+class PlayMonitor(xbmc.Player):
+    def onPlayBackStarted(self):
+        self.setVideoValidity()
+
+    def onAVStarted(self):
+        self.setVideoValidity()
+
+    def setVideoValidity(self):
+        if not self.isPlayingVideo():
+            return
+
+        valid = ''
+        try:
+            if '://' in self.getPlayingFile():
+                valid = 'VIDEO'
+        except RuntimeError:  # Not playing a file
+            pass
+
+        xbmcgui.Window(10000).setProperty(
+            'script.module.youtube.dl_VALID', valid)
+
+
+def showOptions(main=None):
+    w = OptionsDialog(
+        'script-module-youtube-dl-options_dialog.xml',
+        util.ADDON.getAddonInfo('path'), 'main', '720p', main=main
+    )
+    w.doModal()
+    del w
+
+
+class OptionsDialog(xbmcgui.WindowXMLDialog):
+    def __init__(self, *args, **kwargs):
+        self.main = kwargs.get('main')
+        self.player = PlayMonitor()
+        self.player.setVideoValidity()
+
+    def onClick(self, controlID):
+        if controlID == 200:
+            self.main.stopDownload()
+        elif controlID == 201:
+            self.main.stopAllDownloads()
+        elif controlID == 202:
+            self.main.manageQueue()
+        elif controlID == 203:
+            self.main.downloadPlaying()
+        elif controlID == 204:
+            self.main.settings()
+
+
+class main():
+    def __init__(self):
+        arg = self.getArg()
+        if arg == 'INFO':
+            self.showInfo()
+        elif arg == 'UPDATE':
+            self.update()
+        else:
+            showOptions(self)
+
+    def getArg(self):
+        return sys.argv[-1]
+
+    def downloadPlaying(self):
+        title = xbmc.getInfoLabel('Player.Title')
+        # xbmc.getInfoLabel('Player.Filenameandpath')
+        url = xbmc.Player().getPlayingFile()
+        thumbnail = xbmc.getInfoLabel('Player.Art(thumb)')
+        extra = None
+        if '|' in url:
+            url, extra = url.rsplit('|', 1)
+            url = url.rstrip('?')
+        import time
+        info = {'url': url, 'title': title, 'thumbnail': thumbnail,
+                'id': int(time.time()), 'media_type': 'video'}
+        if extra:
+            try:
+                import urlparse
```

Comment:
Py3 compatible?

Reply:
No, however, this commit is for the Gotham branch, probably best a future PR with py3 is made to the Leia branch.